test/event/index.js

```diff
@@ -10,8 +10,44 @@ var Event = require('../../src/event');
 describe('Event - Tests', function () {
   beforeEach(function() {
     Event.removeAllListeners();
-    // TODO consider if this should be a formal API instead
-    Event.observeImmediate_ = {};
+  });
+  it('should remove all listeners', function(done) {
```

Comment:
Test both:
 - standard `on` <= `emit` call sequence
 - unconventional `emit` => `on` sequence
  - i.e. typically we'd see that if there are no listeners an event is not emitted. However, we can see cases, such as slow net connections or with throttling net, an event may be emitted before the listener is added with `on`.
    <p>A case in point is the `POST_AUCTION_RUN` event listener positioned close to the closing `</body>` element and the pubfood auction runs in the header; it has been seen and tested that with throttling enabled, the event is emitted before the listener is added.